tests/FilesystemTest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Behat Gherkin Parser.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Tests\Behat\Gherkin;
+
+use Behat\Gherkin\Exception\FilesystemException;
+use Behat\Gherkin\Filesystem;
+use PHPUnit\Framework\TestCase;
+
+final class FilesystemTest extends TestCase
```

Comment:
Somehow I'm not a fan of creating a new test class for this, but unfortunately the real-world cases are difficult to test (e.g. the `file_exists() ... filemtime()` case).

While it's possible to hook into vfsStream and do some fancy self-destroying files, it's not trivial and it would complicate tests.